### ISSUE TITLE
Add a systemd socket unit and adjust `AmbientCapabilities=`

### ISSUE THREAD

**Issue by eriksjolund** [caddyserver/dist#124]
Now that Caddy supports _socket activation_ adjust the systemd service examples

init/caddy.service
init/caddy-api.service

to make use of a _socket unit_.

I haven't verified this but I'm speculating that we could remove the line

https://github.com/caddyserver/dist/blob/34f51d87d16c3da70156edffdf7c52a6916f48e6/init/caddy.service#L32

`CAP_NET_ADMIN` was added in 

* https://github.com/caddyserver/dist/pull/98

There is a systemd directive `ReceiveBuffer` that looks useful:

https://www.freedesktop.org/software/systemd/man/latest/systemd.socket.html#ReceiveBuffer=

(I don't know if the systemd directive `ReceiveBuffer=` would provide enough functionality to justify the removal of  `CAP_NET_ADMIN`)

**Comment by francislavoie** [comment]
I don't think we can remove any capabilities like that. Socket activation is still an opt-in feature that users have to use in their configs. I don't envision that changing either.

**Comment by eriksjolund** [comment]
Overriding the systemd config is fine with me.
Should I close this issue?
